sematic/examples/bert_fine_tune_scam/pipeline.py

```diff
@@ -0,0 +1,275 @@
+from torch.utils.data.dataset import Dataset
+
+# Sematic
+import sematic
+from sematic import KubernetesResourceRequirements, ResourceRequirements
+from sematic.examples.bert_fine_tune_scam.train_eval import SpamClassifier, train, evaluate, test
+
+@dataclass
+class SpamDataset:
+    data : pd.DataFrame
+
+@sematic.func(inline=True)
+def load_spam_dataset(data_path: str) -> SpamDataset:
+    df = pd.read_csv(data_path)
+    return SpamDataset(data=df)
+
+@dataclass
+class DataLoaderConfig:
+    batch_size: Optional[int] = 32
+
+@sematic.func(inline=True)
+def get_dataloader(dataset: Dataset, config: DataLoaderConfig) -> DataLoader:
+    return DataLoader(dataset, batch_size=config.batch_size)
+
+
+@dataclass
+class TrainConfig:
+    learning_rate: float = 1.0
+    epochs: int = 14
+    max_seq_len: int = 25
+    batch_size: int = 32
+    dry_run: bool = False
+    log_interval: int = 50
+    cuda: bool = False
+    
+
+
+@dataclass
+class PipelineConfig:
+    dataloader_config: DataLoaderConfig
+    train_config: TrainConfig
+    use_cuda: bool = False
+    data_path: str = './data/spam_data.csv'
+
+@sematic.func(inline=True)
+def train_model(
+    config: TrainConfig,
+    train_loader: DataLoader,
+    val_loader: DataLoader,
+    device: torch.device,
+    train_labels: List[object]
+) -> nn.Module:
+    """ Train the model """
+
+    bert, _ = load_bert_model_tokenizer()
+
+    # Freeze all the parameters of bert
+    for param in bert.parameters():
+        param.requires_grad = False
+
+    # Spam Classifier Model
+    model = SpamClassifier(bert=bert).to(device)
+
+    class_wts = compute_class_weight(class_weight = "balanced", classes = np.unique(train_labels), y = train_labels)
+    weights= torch.tensor(class_wts,dtype=torch.float)
+    weights = weights.to(device)
+
+    # Optimizer
+    optimizer = AdamW(model.parameters(), lr = 1e-3)
+    cross_entropy = nn.NLLLoss(weight=weights)
+    
+    # Set initial loss to inf
+    best_valid_loss = float('inf')
+
+    # Store training and validation loss for each epochs
+    train_losses = []
+    valid_losses = []
+
+    for epoch in range(1, config.epochs + 1):
+        
+        # Train Model
+        train_loss, _ = train(
+            model,
+            device,
+            train_loader,
+            optimizer,
+            cross_entropy,
+            epoch,
+            config.log_interval,
+            config.dry_run
+        )
+
+        # Evaluate Model
+        valid_loss, _ = evaluate(
+            model,
+            device,
+            cross_entropy,
+            val_loader
+        )
+
+        if valid_loss < best_valid_loss:
+            best_valid_loss = valid_loss
+            torch.save(model.state_dict(), 'saved_weights.pt')
+        
+
+        train_losses.append(train_loss)
+        valid_losses.append(valid_loss)
+
+        print(f'Training Loss : {train_loss:.2f}')
+        print(f'Validation Loss : {valid_loss:.2f}')
```

Comment:
Use a logger.

At the top of the file below imports add
```python
logger = logging.getLogger(__name__)
```
and use it here instead of `print`.